ros/src/waypoint_updater/waypoint_updater.py

```diff
@@ -33,23 +34,43 @@ def __init__(self):
 
         rospy.Subscriber('/current_pose', PoseStamped, self.pose_cb)
         rospy.Subscriber('/base_waypoints', Lane, self.waypoints_cb)
+        rospy.Subscriber('/traffic_waypoint', Int32, self.traffic_cb)
+        rospy.Subscriber('/obstacle_waypoint', Lane, self.obstacle_cb)
 
-        # TODO: Add a subscriber for /traffic_waypoint and /obstacle_waypoint
-
-        self.final_waypoints_pub = rospy.Publisher('final_waypoints', Lane,
-                                                   queue_size=1)
+        self.pub = rospy.Publisher('final_waypoints', Lane, queue_size=1)
 
         # TODO: Add other member variables you need below
+        self.world = None
```

Comment:
Could they have a more descriptive name? Not trivial to me what "self.world" and "self.ego" are by just looking at this part of the code.

Reply:
Could you suggest something better? Bear in mind that they have quite a few nested fields and shorter names are a advisable.